packages/ui/src/views/unified-pipeline-studio/components/stage-config/unified-pipeline-studio-stage-config-form.tsx

```diff
@@ -119,30 +119,32 @@ export const UnifiedPipelineStudioStageConfigForm = (props: UnifiedPipelineStudi
               <Button variant={'outline'}> Use Template</Button>
             </ButtonGroup>
           </Header>
-          <Inner>
+          <Body>
             <EntityFormLayout.Form>
               <RenderForm className="space-y-6" factory={inputComponentFactory} inputs={stageFormDefinition} />
             </EntityFormLayout.Form>
-          </Inner>
+          </Body>
           <Footer>
-            <ButtonGroup>
-              <Button onClick={() => rootForm.submitForm()}>Submit</Button>
-              <Button variant="secondary" onClick={requestClose}>
-                Cancel
-              </Button>
+            <ButtonGroup className="justify-between">
+              <ButtonGroup>
+                <Button onClick={() => rootForm.submitForm()}>Submit</Button>
+                <Button variant="secondary" onClick={requestClose}>
+                  Cancel
+                </Button>
+              </ButtonGroup>
+              {!!editStageIntention && (
+                <Button
```

Comment:
Same here. Not part of this PR's changes, so not a blocker, but worth doing if you're in the area.